code/game/objects/items/devices/scanners.dm

```diff
@@ -586,47 +590,9 @@ GENE SCANNER
 	return BRUTELOSS
 
 /obj/item/analyzer/attack_self(mob/user)
+	..()
```

Comment:
You can actually get rid of this `attack_self()` implementation wholesale since it's handled by the signal